### DISCUSSION TITLE
[ERROR] Error [ERR_REQUIRE_ESM]: require() of ES Module /root/ws/docusaurus/node_modules/remark-math/index.js from /root/ws/docusaurus/docusaurus.config.js not supported.

### DISCUSSION THREAD

**Discussion by 668168** [facebook/docusaurus#8185] [Q&A] [answered]
# WHAT I WANT
add latex suppot

# WHAT I DO
follow the link step by step
https://docs.theochu.com/docusaurus/latex/

<img width="664" alt="image" src="https://user-images.githubusercontent.com/1046701/194481007-3bb33183-c9f8-4a11-8229-cf4739b932d2.png">

<img width="327" alt="image" src="https://user-images.githubusercontent.com/1046701/194480921-313d11df-d50d-4025-9d13-087f80ebf598.png">

# WHAT HAPPENED
```
[ERROR] Error [ERR_REQUIRE_ESM]: require() of ES Module /root/ws/docusaurus/node_modules/remark-math/index.js from /root/ws/docusaurus/docusaurus.config.js not supported.
Instead change the require of index.js in /root/ws/docusaurus/docusaurus.config.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/root/ws/docusaurus/docusaurus.config.js:6:14)
    at module.exports (/root/ws/docusaurus/node_modules/import-fresh/index.js:32:59)
    at loadSiteConfig (/root/ws/docusaurus/node_modules/@docusaurus/core/lib/server/config.js:36:55)
    at async loadContext (/root/ws/docusaurus/node_modules/@docusaurus/core/lib/server/index.js:31:63)
    at async Command.build (/root/ws/docusaurus/node_modules/@docusaurus/core/lib/commands/build.js:51:21)
[INFO] Docusaurus version: 2.1.0
Node version: v16.14.2
```

# WHAT SHOULD I DO?

**Comment by slorber** [comment, marked as answer]
Hi

I suspect your remark-math version to be incorrect, despite what your package.json screenshot shows

You have to use the recommended versions, exactly like we do on our site because Docusauurs is not compatible with newer versions yet unless you trick a bit the system

https://docusaurus.io/docs/markdown-features/math-equations#upgrading-rehype-katex-beyond-recommended-version

---

without a repro, it's impossible to help you and double-check your install => docusaurus.new => show us the problem in action

**Reply by 668168** [reply]
Hi, Slorber,
Thanks for your advice.
As you said "without a repro, it's impossible to help you and double-check your install => docusaurus.new => show us the problem in action"

STEP BY STEP:
1. reinstall a new docusaurus
2. add latex support follow the link:   https://docs.theochu.com/docusaurus/latex/
3. same error happend.

ATTACHMENT:
1. the all file of the new docusaurus, just modify the docusaurus.config.js package.json
https://github.com/shuivin/docusaurus-debug

any advice ?

**Reply by slorber** [reply]
Read the official doc, not third-party docs that may be outdated

Your package-lock.json doesn't even contain remark-math nor rehype-katex so even if I run your code locally and it works for me, you may have a local problem because you could have the wrong versions installed locally. If you need to, open node_modules and be sure you have the correct versions

**Reply by 668168** [reply]
Thanks for your help Slorber ! 
I resolve the issue .
as your said , the root cause is remark-math version. 
blow is the step, hope useful for others:
1. I found the remark-math version is ^5.1.1 in my-website/node_modules/remark-math# vim package.json
2. I MUST DELETE the folder my-website/node_modules/remark-math
3. modify the my-website/package.json , set the  "remark-math": "^3.0.1",
4. npm install remark-math
DONE
